test/powershell/Host/TabCompletion/TabCompletion.Tests.ps1

```diff
@@ -152,6 +152,31 @@ Describe "TabCompletion" -Tags CI {
         $res.CompletionMatches[1].CompletionText | Should -BeExactly '-Functionality'
     }
 
+    It 'Should work for variable assignment of enums: <inputStr>' -TestCases @(
+        @{ inputStr = '$ErrorActionPreference = '; filter = ''; doubleQuotes = $false }
+        @{ inputStr = '$ErrorActionPreference='; filter = ''; doubleQuotes = $false }
+        @{ inputStr = '$ErrorActionPreference="'; filter = ''; doubleQuotes = $true }
+        @{ inputStr = '$ErrorActionPreference = ''s'; filter = '| Where-Object { $_ -like "''s*" }'; doubleQuotes = $false }
+        @{ inputStr = '$ErrorActionPreference = "siL'; filter = '| Where-Object { $_ -like ''"sil*'' }'; doubleQuotes = $true }
```

Comment:
We should also have tests on a variable that isn't special in some way:

```powershell
enum Animal { Duck; Goose; Horse; Giraffe }

[Animal]$animal = '
```

Reply:
will add